moveos/moveos/src/moveos.rs

```diff
@@ -399,12 +519,33 @@ impl MoveOS {
             .add(tx_result)
             .expect("Add tx_result to TxContext should always success");
 
+        if let Some(pay_gas) = can_pay_gas {
+            if pay_gas {
+                let verified_action = &action_opt_cloned.clone().unwrap();
+                let func_call_opt = {
+                    match verified_action {
+                        VerifiedMoveAction::Function { call } => Some(call),
+                        _ => None,
+                    }
+                };
+
+                if let Some(func_call) = func_call_opt {
+                    let module_id = func_call.function_id.module_id.address();
+                    session.storage_context_mut().tx_context.sender = *module_id;
```

Comment:
Directly changing the `tx_context.sender` to `module_address` is not a good idea. We can put a variable, such as `GasPaymentAccount`, to the tx_context's map, and get it in the Move Code to charge the gas.